pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -339,41 +358,59 @@ func (vm *windows) initializeBootstrapperFiles() error {
 }
 
 // createService creates the service on the Windows VM
-func (vm *windows) createService(svc service) error {
-	out, err := vm.Run("sc.exe create "+svc.Name()+" binPath=\""+svc.BinaryPath()+" "+
-		svc.Args()+"\" start=auto", false)
+func (vm *windows) createService(svc *service) error {
+	out, err := vm.Run("sc.exe create "+svc.name+" binPath=\""+svc.binaryPath+" "+
+		svc.args+" start=auto", false)
 	if err != nil {
 		return errors.Wrapf(err, "failed to create service with output: %s", out)
 	}
 	return nil
 }
 
 // stopService stops the service that was already running
-func (vm *windows) stopService(svc service) error {
-	out, err := vm.Run("sc.exe stop "+svc.Name(), false)
+func (vm *windows) stopService(svc *service) error {
+	out, err := vm.Run("sc.exe stop "+svc.name, false)
 	if err != nil {
-		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.Name(), out)
+		return errors.Wrapf(err, "failed to stop %s service with output: %s", svc.name, out)
 	}
 	return nil
 }
 
-// isRunning gets the status of given service
-func (vm *windows) isRunning(svc service) bool {
-	out, err := vm.Run("sc.exe query "+svc.Name(), false)
+// serviceExists checks if the given service exists on Windows VM
+func (vm *windows) serviceExists(serviceName string) (bool, error) {
+	_, err := vm.Run("sc.exe qc "+serviceName, false)
 	if err != nil {
-		log.Info("failed to query service", "service", svc.Name(), "output", out, "err", err)
-		return false
+		if strings.Contains(err.Error(), "status 1060") {
+			return false, nil
+		}
+		return false, err
 	}
-	return strings.Contains(out, "RUNNING")
+	return true, nil
+}
+
+// isRunning checks the status of given service
+func (vm *windows) isRunning(serviceName string) (bool, error) {
+	out, err := vm.Run("sc.exe query "+serviceName, false)
+	if err != nil {
+		return false, err
+	}
+	return strings.Contains(out, "RUNNING"), nil
 }
 
 // startService starts a previously created Windows service
-func (vm *windows) startService(svc service) error {
-	out, err := vm.Run("sc.exe start "+svc.Name(), false)
+func (vm *windows) startService(svc *service) error {
+	serviceRunning, err := vm.isRunning(svc.name)
```

Comment:
nit: Checking for the current state should be responsibility of caller method.

Reply:
as discussed in the discussion earlier I am avoiding this here to avoid code repetition in both service caller methods.